pubtools/pulplib/_impl/client/client.py

```diff
@@ -181,6 +185,50 @@ def search_repository(self, criteria=None):
             response_f, lambda data: self._handle_page(Repository, search, data)
         )
 
+    def get_maintenance_report(self):
+        """Get the current maintenance mode status for this Pulp server.
+
+        Returns:
+            Future[:class:`~pubtools.pulplib.MaintenanceReport`]
+                A future describes the maintenance status
+
+        .. versionadded:: 1.4.0
+        """
+        report_ft = self._do_get_maintenance()
+
+        return f_map(
+            report_ft,
+            lambda data: MaintenanceReport._from_data(data)
+            if data
+            else MaintenanceReport(),
+        )
+
+    def set_maintenance(self, report):
+        """Set maintenance mode for this Pulp server.
+
+        Args:
+            report:
+                An updated :class:`~pubtools.pulplib.MaintenanceReport` object that
+                will be used as the newest maintenance report.
+        Return:
+            Future[list[:class:`~pubtools.pulplib.Task`]]
+                A future which is resolved when maintenance mode has been updated successfully.
+
+                The future contains a task triggered and awaited during the publish
+                maintenance repository operation.
+
+        .. versionadded:: 1.4.0
+        """
+        report_json = json.dumps(report._export_dict(), indent=4, sort_keys=True)
+        report_fileobj = StringIO(report_json)
+
+        repo = self.get_repository("redhat-maintenance").result()
```

Comment:
As an aside (not arguing to change it now), I'm wondering if maybe we should have just let this library also create the repo when it doesn't exist.  I'm not much benefit to the current design where we require some other tool to be run first.